### ISSUE TITLE
ObjectExpressions are parsed as BlockStatements

### ISSUE THREAD

**Issue by thosakwe** [asgerf/parsejs.dart#3]
Willing to submit a PR, but the following:

```js
{ a: 2, b: 'a'}
```

is parsed as a `BlockStatement`, not an `ObjectExpression`.

**Comment by asgerf** [comment]
Hi,

Unfortunately that is how JS [actually works](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#Object_literals). Although most JS interactive consoles has a workaround for it, this does not apply to real JS code being read from a file.

For example, here is an interactive session with node.js:
```bash
~ $ node
> { a: 2, b: 'a'}
{ a: 2, b: 'a' }
```

But if I read it from a file instead:
```bash
~ $ cat foo.js
{ a: 2, b: 'a'}
~ $ node foo.js
foo.js:1
(function (exports, require, module, __filename, __dirname) { { a: 2, b: 'a'}
                                                                       ^
SyntaxError: Unexpected token :
    at createScript (vm.js:53:10)
    at Object.runInThisContext (vm.js:95:10)
    at Module._compile (module.js:543:28)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:427:7)
    at startup (bootstrap_node.js:148:9)
```

If you know you are parsing an expression, and you are willing to rely on internal API, you can call the parser directly, like this:
```dart
new Parser(new Lexer(...)).parseExpression()
```

But I can see the value in parsing a standalone JS expression, so I can expose this functionality in the public API if you like.

**Comment by thosakwe** [comment]
That would be great, thank you! I'm currently using this to write an [interpreter](https://github.com/samurai-dart/samurai), and this would make a REPL more feasible.

**Comment by asgerf** [comment]
I have released version 1.2.0 which adds the named argument `parseAsExpression`.

If you want to support both expressions and statements as input, you can use a wrapper like this:
```dart
Program parseInput(String text) {
  try {
    return parsejs(text, parseAsExpression: true);
  } on ParseError {
    return parsejs(text);
  }
}
```

I am closing this issue, but feel free to reopen if it doesn't work for you.

**Comment by thosakwe** [comment]
Thank you very much! Maybe one of these days I'll submit a PR to support ES6 syntax...
